app/questionnaire/rules.py

```diff
@@ -124,22 +126,21 @@ def get_date_match_value(date_comparison, answer_store, schema, metadata):
     elif "meta" in date_comparison:
         match_value = get_metadata_value(metadata, date_comparison["meta"])
 
-    match_value = convert_to_datetime(match_value)
+    if match_value:
+        match_value = convert_to_datetime(match_value)
 
-    if "offset_by" in date_comparison and match_value:
-        offset = date_comparison["offset_by"]
-        match_value = match_value + relativedelta(
-            days=offset.get("days", 0),
-            months=offset.get("months", 0),
-            years=offset.get("years", 0),
-        )
+        if "offset_by" in date_comparison and match_value:
+            offset = date_comparison["offset_by"]
+            match_value = match_value + relativedelta(
+                days=offset.get("days", 0),
+                months=offset.get("months", 0),
+                years=offset.get("years", 0),
+            )
 
     return match_value
 
 
-def convert_to_datetime(value: Optional[str]) -> Optional[datetime]:
-    if not value:
-        return None
+def convert_to_datetime(value: str) -> datetime:
```

Comment:
The typing led me there, but it bugged me the function does the falsy-value check rather than the check being done by the callers of the function. I did follow the code through and check usages and tests, and they all ran fine after the changes. I can revert and just overload if that's the preference

Reply:
Given the change has been done already and references refactored, happy to keep as is 👍

Reply:
I reverted it as it happens - I had broken a functional test and see this (https://github.com/ONSdigital/eq-questionnaire-runner/pull/665#discussion_r721999745) it isn't trivial to fix